addons/web_editor/static/src/components/media_dialog/image_selector.js

```diff
@@ -187,10 +187,27 @@ export class ImageSelector extends FileSelector {
         });
     }
 
-    validateUrl(...args) {
+    async validateUrl(...args) {
         const { isValidUrl, path } = super.validateUrl(...args);
-        const isValidFileFormat = IMAGE_EXTENSIONS.some(format => path.endsWith(format));
-        return { isValidFileFormat, isValidUrl };
+        const isValidFileFormat = isValidUrl && await new Promise(resolve => {
+            const img = new Image();
+            img.src = path;
+            img.onload = () => resolve(true);
+            img.onerror = async () => {
+                try {
+                    const response = await fetch(path, {method: 'HEAD'});
+                    if (response.ok) {
+                        const contentType = response.headers.get('content-type');
+                        resolve(contentType && IMAGE_EXTENSIONS.includes(contentType));
+                    } else {
+                        resolve(false)
+                    }
```

Comment:
I'm not sure I undertand what happens here. What is the practical use case where the image could not be loaded by the browser but we still expect to have a valid image when doing a HEAD request on the url?

Reply:
Seeing this again, I agree. If the image cannot be loaded by the browser, there’s no reason to expect it to be valid after a HEAD request. I have updated the code accordingly.